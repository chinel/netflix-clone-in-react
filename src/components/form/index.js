import React from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};
