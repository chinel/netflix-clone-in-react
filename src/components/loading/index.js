import React from "react";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`}
    </Spinner>
  );
}
