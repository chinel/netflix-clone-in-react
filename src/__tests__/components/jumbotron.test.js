import React from "react";
import { render } from "@testing-library/react";
import { Jumbotron } from "../../components";
import JumboData from "../../fixtures/jumbo";

describe("<Jumbotron/>", () => {
  it("renders the <Jumbotron /> with populated data", () => {
    const { container, getByText, getByAltText, getTestId } = render();
  });
});
