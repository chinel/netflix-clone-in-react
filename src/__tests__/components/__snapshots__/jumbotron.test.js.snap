// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron/> renders the <Jumbotron /> with populated data 1`] = `
<div
  class="jumbotron__Container-rhnsqh-7 kOVxtT"
>
  <div
    class="jumbotron__Item-rhnsqh-0 kJombk"
  >
    <div
      class="jumbotron__Inner-rhnsqh-1 ckmfPJ"
      direction="row"
    >
      <div
        class="jumbotron__Pane-rhnsqh-2 kpHiLd"
      >
        <h1
          class="jumbotron__Title-rhnsqh-3 dLfMfV"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="jumbotron__SubTitle-rhnsqh-4 cdSsOm"
        >
          Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-rhnsqh-2 kpHiLd"
      >
        <div
          class="jumbotron__ImageCon-rhnsqh-5 ewNFDU"
          data-testid="1-jumbo-image"
        >
          <img
            alt="Tiger King on Netflix"
            class="jumbotron__Image-rhnsqh-6 isulmV"
            src="/images/misc/home-tv.jpg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-rhnsqh-0 kJombk"
  >
    <div
      class="jumbotron__Inner-rhnsqh-1 ckmfPJ"
      direction="row"
    >
      <div
        class="jumbotron__Pane-rhnsqh-2 kpHiLd"
      >
        <h1
          class="jumbotron__Title-rhnsqh-3 dLfMfV"
        >
          Download your programmes to watch on the go.
        </h1>
        <h2
          class="jumbotron__SubTitle-rhnsqh-4 cdSsOm"
        >
          Save your data and watch all your favourites offline.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-rhnsqh-2 kpHiLd"
      >
        <div
          class="jumbotron__ImageCon-rhnsqh-5 ewNFDU"
          data-testid="2-jumbo-image"
        >
          <img
            alt="Watch on mobile"
            class="jumbotron__Image-rhnsqh-6 isulmV"
            src="/images/misc/home-mobile.jpg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-rhnsqh-0 kJombk"
  >
    <div
      class="jumbotron__Inner-rhnsqh-1 ckmfPJ"
      direction="row"
    >
      <div
        class="jumbotron__Pane-rhnsqh-2 kpHiLd"
      >
        <h1
          class="jumbotron__Title-rhnsqh-3 dLfMfV"
        >
          Watch everywhere.
        </h1>
        <h2
          class="jumbotron__SubTitle-rhnsqh-4 cdSsOm"
        >
          Stream unlimited films and TV programmes on your phone, tablet, laptop and TV without paying more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-rhnsqh-2 kpHiLd"
      >
        <div
          class="jumbotron__ImageCon-rhnsqh-5 ewNFDU"
          data-testid="3-jumbo-image"
        >
          <img
            alt="Money Heist on Netflix"
            class="jumbotron__Image-rhnsqh-6 isulmV"
            src="/images/misc/home-imac.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
